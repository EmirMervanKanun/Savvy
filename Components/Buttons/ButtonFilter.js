import React, { useState } from 'react';
import { View, Pressable, StyleSheet,Text } from 'react-native';
import TitelAmountText from '../TextComponents/TitelAmountText';
import COLORS from '../Colors';

const ButtonFilter = ({selected, onPress, label}) => {
  return (
      <Pressable style={styles.container} onPress={onPress}>
        <View style={selected? {borderBottomWidth:2, borderColor: COLORS.schriftDark} : null}>
          <Text style={styles.filterText}><TitelAmountText>{label}</TitelAmountText></Text>
        </View>
          
      </Pressable>
  );
};

const FilterGroup = ({ options, changedFilter, selectedAtStart}) => {
  const [selectedOption, setSelectedOption] = useState(selectedAtStart || null);

  return (
    <View style={styles.groupContainer}>
      {options.map((option, index) => (
        <ButtonFilter
          key={index}
          label={option}
          selected={selectedOption === option}
          onPress={(filter) => {
            setSelectedOption(option);
            changedFilter = option;
          }
        }/>
      ))}
    </View>
  );
}; export default FilterGroup;

const styles = StyleSheet.create({
  groupContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    gap: 14,
  },
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  underlined: {
    textDecorationLine: 'underline',
  },
  filterText: {
    color: COLORS.schriftDark,
    fontSize: 16,
    fontWeight: 'bold',
  },
});


