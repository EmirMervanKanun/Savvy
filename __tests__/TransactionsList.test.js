import React from 'react';
import renderer, { act } from 'react-test-renderer';
import TransactionScreen from '../Screens/Transactions/TransactionsList';
import ButtonFilter from '../Components/Buttons/ButtonFilter';
import TransactionsSwipeList from '../Components/Objects/TransactionSwipeList';

// Mocken der ButtonFilter-Komponente
jest.mock('../Components/Buttons/ButtonFilter', () => ({ options, changedFilter, selectedAtStart }) => (
  <div data-testid="mocked-button-filter" options={options} changedFilter={changedFilter} selectedAtStart={selectedAtStart} />
));

// Mocken der TransactionsSwipeList-Komponente
jest.mock('../Components/Objects/TransactionSwipeList', () => ({ transactions }) => (
  <div data-testid="mocked-transaction-swipe-list" transactions={transactions} />
));

describe('TransactionScreen Komponente', () => {
  let component;

  beforeEach(() => {
    act(() => {
      component = renderer.create(<TransactionScreen />);
    });
  });

  it('rendert korrekt', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('überprüft den Filterwechsel in ButtonFilter', () => {
    act(() => {
      const mockedButtonFilter = component.root.findByProps({ 'data-testid': 'mocked-button-filter' });
      // Überprüfen, ob changedFilter als Funktion existiert, bevor Sie sie aufrufen
      if (mockedButtonFilter.props.changedFilter) {
        mockedButtonFilter.props.changedFilter('Einnahmen');
      }
    });

  });
});
