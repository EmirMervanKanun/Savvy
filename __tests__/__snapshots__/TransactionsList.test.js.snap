// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionScreen Komponente rendert korrekt 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "display": "flex",
          "gap": 32,
          "paddingBottom": 220,
          "paddingTop": 24,
        }
      }
    >
      <div
        data-testid="mocked-button-filter"
        options={
          [
            "Gesamt",
            "Einnahmen",
            "Ausgaben",
          ]
        }
        selectedAtStart="Gesamt"
      />
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "column",
            "paddingTop": 8,
          }
        }
      >
        <div
          data-testid="mocked-transaction-swipe-list"
          transactions={
            [
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": false,
                  "isStan": false,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": false,
                  "isStan": false,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
              {
                "details": {
                  "amount": 1000,
                  "date": "01.01.2021",
                  "img": 1,
                  "isIncome": true,
                  "isStan": true,
                  "stan": "Monthly",
                },
                "title": "Transaction1",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
